packages/dynamite/dynamite/lib/src/builder/client.dart

```diff
@@ -275,8 +274,8 @@ Iterable<Method> buildTags(
           ),
         );
 
-        buildPatternCheck(result, parameter).forEach(code.writeln);
         buildParameterSerialization(result, parameter).forEach(code.writeln);
+        buildPatternCheck(result, parameter).forEach(code.writeln);
```

Comment:
Why is the order switched? I think we should first check if the parameter values match the patterns and then start to use them to construct the path.